### DISCUSSION TITLE
How do I read SNBT files?

### DISCUSSION THREAD

**Discussion by GoldenFurredFox** [FTBTeam/FTB-Mods-Issues#1304] [Q&A]
To be clear, I'm not referring to actually reading it in the human way, I'm referring to the read/write capabilities of a computer.

The file can't be opened in a standard NBT explorer file, so I need to know how to open it. I don't even care about writing/erasing data to/from it, I just want to read the contents so I can find a user's old claim.

**Comment by desht** [comment]
There are methods in the `SNBT` class in FTB Library which will read a .snbt file into a `CompoundTag`, or write as `CompoundTag` to a .snbt file.

But if you're just trying to find a user's old claim, snbt files as saved by FTB Chunks are human-readable. You'll need to know the user's Minecraft UUID (which you can easily get from sites like https://mcuuid.net/). The chunks are stored in `world/ftbchunks/<uuid>.snbt`, in the `chunks` field - that is a list of chunk coordinates (one chunk = 16 blocks), and the time at which they were claimed.

If the user was in a party team, the chunks will be in `world/ftbchunks/<party-uuid>.snbt` instead. That file will have a `member_data` section including the UUID of each member.

**Reply by GoldenFurredFox** [reply]
I gathered all of this information from the wiki already. As I said in my initial post, I'm not asking if they're human-readable. In computer terms, I'm literally just asking how to open the file so that I can read it to begin with.

**Reply by MichaelHillcox** [reply]
It's very close to the default SNBT format minecraft uses but if you want to parse them by a system. You'd either need to pull our library mod into your dev-env or you'd need to write your own parser for SNBT or modify the spec of an Minecraft SNBT parser
